pageserver/src/tenant/storage_layer/image_layer.rs

```diff
@@ -154,6 +157,7 @@ pub struct ImageLayerInner {
 
     /// Reader object for reading blocks from the file.
     file: FileBlockReader,
+    vectored_blob_reader: VectoredBlobReader,
```

Comment:
Hmm, is VectoredBlobReader carrying the same VirtualFile as FileBlockReader?

This isn't a huge overhead, but layers are very numerous so if there's some way to avoid  the duplication it might be worth it.

Reply:
Done in https://github.com/neondatabase/neon/pull/6780/commits/0804e0c776b12e841861d53e42f646b5ce92a17d